drafts/bitcartel-payment-disclosure/draft.rst

```diff
@@ -0,0 +1,345 @@
+
+To mitigate against a replay attack, the recipient could enforce a policy where the first person to present a given payment disclosure is the honest sender.
+
+[TODO: Example?] To prevent a man-in-the-middle attack, the recipient could pose an interactive challenge involving some out-of-band secret which only the sender would pass successfully.
+
+A non-interactive solution is also possible:
+
+#. The sender includes data, such as a refund address, in the memo field of a payment to a shielded address.  The memo field is non-malleable as it is part of the data signed by the sender when creating JoinSplitSig [PROTOCOL] §4.1.6 Signatures.  Currently, the public key to verify the signature is stored in the blockchain along with the transaction.
+#. When presenting the payment disclosure to a recepient, the sender signs the payment disclosure with the JoinSplitSig private key which only they would know.  The JoinSplitSig key pair is ephemeral and normally discarded but the method described in this ZIP will retain the private portion.  The recipient can then use the public key to verify the signed disclosure.
+
+Using the JoinSplitSig key pair in this way also means that a sender can make payments without having to first provide a refund address or other details at the time of the original payment.  If at a later date the sender and recipient need to confirm the payment, the sender can sign and send any data required to the recipient, who can in turn verify it.
+
+
+Known Issues
+------------
+Both plaintext outputs of a joinsplit are currently encrypted with symmetric keys derived from the same ephemeral secret key, as discusshed here: https://github.com/zcash/zcash/issues/558#issuecomment-167819936
+
+This means that a payment disclosure which includes the ephemeral secret key intended to decipher the note ciphertext belonging to a particular joinsplit output, creates a privacy leak by also deciphering the other joinsplit output and revealing it's amount and payment address.
+
+Note the current implementation of the ``z_sendmany`` RPC call:
+
+- When sending from a transparent address to shielded addresses, each shielded address will occupy one output of a joinsplit.  Thus a payment address to a different recipient could be revealed.
+
+- When sending involves chained joinsplits, for each joinsplit, one output is for payment to a recipient's address, while the other output is used as change back to the sender.  Thus the change address will be revealed, which is the sender's own shielded address.
+
+A proposal to prevent information leakage of change addresses is currently under development [KDFT].  TODO: Does this block ZIP approval?
+
+Also note that an independent third party cannot know for sure if the sender and recipient are not colluding to hide value transfer.  JoinSplits have two inputs and two outputs.  The recipient may identify their receiving address for one output, but not disclose the fact that the other output also belongs to them.
```

Comment:
s/for sure if/for sure that/

Reply:
This seems to be moot because the sender and recipient could just as well transfer value in a different joinsplit without disclosing it.

Reply:
Updated note with this.